src/features/content-answers/ContentAnswersModal.tsx

```diff
@@ -53,11 +54,15 @@ const ContentAnswersModal = ({ api, currentUser, file, isOpen, onRequestClose }:
         setQuestions([...updatedQuestions, lastQuestion]);
     }, []);
 
-    const handleErrorCallback = useCallback((error: ElementsXhrError): void => {
+    const handleErrorCallback = useCallback((error: ElementsXhrError, updatedQuestions): void => {
         setIsLoading(false);
-        if (error) {
-            // TODO: render error component
+
+        const lastQuestion = updatedQuestions.pop();
+        if (lastQuestion && lastQuestion.prompt) {
+            lastQuestion.error = error;
         }
+
+        setQuestions([...updatedQuestions, lastQuestion]);
```

Comment:
should we still update state if there isnt a `lastQuestion`?

Reply:
It should be guaranteed that there is always a lastQuestion in the question array. The reason is that whenever we submit a question, the new question object with the prompt only will be appended into the question array.